### PR TITLE
koordlet: skip fillExtensionMap if podMetric is nil

### PR DESCRIPTION
**Pull request by jasonliu747**
Signed-off-by: Jason Liu <jasonliu747@gmail.com>

### V. Checklist

- [ ] I have written necessary docs and comments
- [ ] I have added necessary unit tests and integration tests
- [ ] All checks passed in `make test`

### CODE REVIEW

File: pkg/koordlet/statesinformer/states_nodemetric.go [hunk 1]
```diff
@@ -330,8 +330,8 @@ func (r *nodeMetricInformer) collectMetric() (*slov1alpha1.NodeMetricInfo, []*sl
 	}
 	for _, podMeta := range podsMeta {
 		podMetric := r.collectPodMetric(podMeta, podQueryParam)
-		r.fillExtensionMap(podMetric, podMeta.Pod)
 		if podMetric != nil {
+			r.fillExtensionMap(podMetric, podMeta.Pod)
```
Comment (saintube) [inline pkg/koordlet/statesinformer/states_nodemetric.go:334]: could you please add a unit test

Comment (jasonliu747) [reply pkg/koordlet/statesinformer/states_nodemetric.go:334]: I could, but it's an empty function. 😂😂😂

**Review by jasonliu747 — COMMENTED** [review]
/approve

**Pull request merged**